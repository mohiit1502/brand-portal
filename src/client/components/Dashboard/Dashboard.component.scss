.c-Dashboard {
  overflow-y: auto;
}
