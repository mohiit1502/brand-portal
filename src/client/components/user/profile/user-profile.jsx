import React from "react";
import { connect } from "react-redux";
import "../../../styles/user/profile/user-profile.scss";
import PropTypes from "prop-types";
import CustomInput from "../../custom-input/custom-input";

class UserProfile extends React.Component {

  constructor (props) {
    super(props);
    this.onInputChange = this.onInputChange.bind(this);
    this.setFormData = this.setFormData.bind(this);
    this.disableInput = this.disableInput.bind(this);
    this.toggleUnderwritingCheck = this.toggleUnderwritingCheck.bind(this);

    this.state = {
      form: {
        isDisabled: true,
        underwritingChecked: false,
        id: "user-profile-form",
        inputData: {
          firstName: {
            label: "First Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          lastName: {
            label: "Last Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          companyName: {
            label: "Company Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          emailId: {
            label: "Email ID",
            required: true,
            value: "",
            type: "email",
            pattern: null,
            disabled: true
          },
          phone: {
            label: "Mobile Number",
            required: false,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          }
        }
      }
    };
  }

  componentDidMount() {
    const obj = {
      firstName: "Shubhansh",
      lastName: "Sahai",
      companyName: "Walmart Labs",
      emailId: "shubhansh.sahai@walmartlabs.com",
      phone: "9686648597"
    };

    this.setFormData(obj);

  }

  setFormData(obj) {
    const form = {...this.state.form};
    form.inputData.firstName.value = obj.firstName;
    form.inputData.firstName.disabled = true;
    form.inputData.lastName.value = obj.lastName;
    form.inputData.lastName.disabled = true;
    form.inputData.companyName.value = obj.companyName;
    form.inputData.companyName.disabled = true;
    form.inputData.emailId.value = obj.emailId;
    form.inputData.emailId.disabled = true;
    form.inputData.phone.value = obj.phone;
    form.inputData.phone.disabled = true;

    this.setState({
      form
    });
  }

  onInputChange (evt, key) {
    if (evt && evt.target) {
      const targetVal = evt.target.value;
      this.setState(state => {
        state = {...state};
        state.form.inputData[key].value = targetVal;
        return {
          ...state
        };
      });
    }
  }

  disableInput (disable) {
    disable = !!disable;

    const form = {...this.state.form};

    form.isDisabled = disable;

    form.inputData.firstName.disabled = disable;
    form.inputData.lastName.disabled = disable;
    form.inputData.companyName.disabled = disable;
    form.inputData.emailId.disabled = disable;
    form.inputData.phone.disabled = disable;

    this.setState({
      form
    });

  }

  toggleUnderwritingCheck () {
    const form = {...this.state.form};
    form.underwritingChecked = !form.underwritingChecked;
    this.setState({form});
  }

  render () {

    return (
      <div className="row user-profile-content h-100">
        <div className="col h-100">
          <div className="row h-10">
            <div className="col">
              <h3>User Profile</h3>
            </div>
          </div>
          <div className="row mt-4 h-90">
            <div className="col">
              <form className="h-100" autoComplete="off">
                <div className="row h-60">
                  <div className="col h-100">
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"firstName"}
                          inputId={"firstName"}
                          formId={this.state.form.id} label={this.state.form.inputData.firstName.label}
                          required={this.state.form.inputData.firstName.required} value={this.state.form.inputData.firstName.value}
                          type={this.state.form.inputData.firstName.type} pattern={this.state.form.inputData.firstName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.firstName.disabled} />
                      </div>
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"lastName"}
                          inputId={"lastName"}
                          formId={this.state.form.id} label={this.state.form.inputData.lastName.label}
                          required={this.state.form.inputData.lastName.required} value={this.state.form.inputData.lastName.value}
                          type={this.state.form.inputData.lastName.type} pattern={this.state.form.inputData.lastName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.lastName.disabled} />
                      </div>
                    </div>
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"companyName"}
                          inputId={"companyName"}
                          formId={this.state.form.id} label={this.state.form.inputData.companyName.label}
                          required={this.state.form.inputData.companyName.required} value={this.state.form.inputData.companyName.value}
                          type={this.state.form.inputData.companyName.type} pattern={this.state.form.inputData.companyName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.companyName.disabled} />
                      </div>
                    </div>
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"emailId"}
                          inputId={"emailId"}
                          formId={this.state.form.id} label={this.state.form.inputData.emailId.label}
                          required={this.state.form.inputData.emailId.required} value={this.state.form.inputData.emailId.value}
                          type={this.state.form.inputData.emailId.type} pattern={this.state.form.inputData.emailId.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.emailId.disabled} />
                      </div>
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"phone"}
                          inputId={"phone"}
                          formId={this.state.form.id} label={this.state.form.inputData.phone.label}
                          required={this.state.form.inputData.phone.required} value={this.state.form.inputData.phone.value}
                          type={this.state.form.inputData.phone.type} pattern={this.state.form.inputData.phone.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.phone.disabled} />
                      </div>
                    </div>
                    <div className="row mb-5 password-reset-row">
                      <div className="col-3">
                        <div className="btn btn-primary btn-sm no-border-radius px-3">
                          Password Reset
                        </div>
                        <span className="badge badge-pill bg-white text-blue ml-2 cursor-pointer"> ? </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div className="row h-40">
                  <div className="col h-100">
                    <div className="row pt-5">
                      <div className="col-xl-6 col-10">
                        <label>
                          <input type="checkbox" className="mr-2" checked={this.state.form.underwritingChecked} onChange={this.toggleUnderwritingCheck}/>
                          I have a good faith belief that the use of the material in the manner complained of is not authorized by the copyright owner, its agent, or the law.
                        </label>
                      </div>
                    </div>
                    <div className="row pt-5">
                      <div className="col-xl-3 col-5" />
                      <div className="col-xl-3 col-5 text-right">
                        {
                          this.state.form.isDisabled ?
                            (
                              <div className="btn btn-primary btn-sm no-border-radius px-4" onClick={() => this.disableInput(false)}>
                                Edit
                              </div>
                            ) :
                            (
                              <span>
                                <div className="btn btn-link font-size-14 px-4 mr-3" onClick={() => this.disableInput(true)}>Cancel</div>
                                <button type="submit" className="btn btn-primary btn-sm no-border-radius px-4 font-size-14" disabled={!this.state.form.underwritingChecked}>
                                  Save
                                </button>
                              </span>
                            )
                        }

                      </div>
                    </div>
                  </div>
                </div>

              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => state;

export default connect(
  mapStateToProps,
  dispatch => ({dispatch})
)(UserProfile);
