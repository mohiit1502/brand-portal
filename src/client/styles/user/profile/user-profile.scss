@import "../../override/variables";

.user-profile-content {
  .password-reset-row {
    .badge {
      border: 1px solid $blue;
    }
  }
}
