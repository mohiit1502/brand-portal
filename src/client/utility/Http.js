/* eslint-disable complexity */
/* eslint-disable no-unused-expressions */
/* eslint-disable max-params */
/* eslint-disable max-statements */
import fetch from "node-fetch";
import queryString from "query-string";
import ClientHttpError from "./ClientHttpError";
import { NOTIFICATION_TYPE } from "../actions/notification/notification-actions";
import CONSTANTS from "../constants/constants";
export default class Http {

  static async get(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});

    const options = {
      method: "GET"
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;

    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message, err.code);
  }

  static async post(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async postAsFormData(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: data,
      headers: { }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async put(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "PUT",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };

    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);
  }

  static async delete(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "DELETE",
      noInject: true
    };
    return fetch(urlString, options);
  }

  displayToast(status, toastcallback) {
    if (new RegExp(CONSTANTS.CODES.ERRORCODES.SERVERERROR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Unable to reach our services currently, please try again in sometime.`);
    } else if (new RegExp(CONSTANTS.CODES.ERRORCODES.FOURNOTFOUR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Requested Resource could not found.`);
    }
  }
}
