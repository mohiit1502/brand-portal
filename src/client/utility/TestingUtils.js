import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const realStore = state => {
  const configureStore = initialState => {
    const composeEnhancers =
      typeof window === "object" &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          serialize: { // prettier-ignore
            immutable: Immutable
          }
        }) : compose;
    const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunk)));

    if (module.hot) {
      module.hot.accept("./reducers", () => {
        const nextRootReducer = require("./reducers").default;
        store.replaceReducer(nextRootReducer);
      });
    }

    return store;
  };

  return configureStore(state || window.__WML_REDUX_INITIAL_STATE__);
}

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
