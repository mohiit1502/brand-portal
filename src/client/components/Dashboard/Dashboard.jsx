// eslint-disable-next-line filenames/match-regex
import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {dispatchFilter} from "../../actions/dashboard/dashboard-actions";
import {DashboardHeader, GenericErrorPage, WidgetContainer} from "../index";
import {showNotification} from "../../actions/notification/notification-actions";
import Http from "../../utility/Http";
import Helper from "../../utility/helper";
import widgetConfig from "./../../config/contentDescriptors/widgets";
import AUTH_CONFIG from "../../config/authorizations";
import "./Dashboard.component.scss";
import mixpanel from "../../utility/mixpanelutils";
import MIXPANEL_CONSTANTS from "../../constants/mixpanelConstants";
import { TOGGLE_ACTIONS, toggleModal } from "../../actions/modal-actions";

class Dashboard extends React.PureComponent {
  constructor(props) {
    super(props);
    const {modalsMeta, toggleModal} = props;
    this.state = {
      data: {
        brand: {},
        claim: {},
        user: {},
        claimsByType: [],
        claimsByBrands: [],
        claimsByUsers: []
      },
      error: false,
      fetchComplete: false
    };
    // toggleModal(TOGGLE_ACTIONS.SHOW, {templateName: "StatusModalTemplate", ...modalsMeta.GO_TO_USER_PROFILE});

  }

  componentDidMount() {
    const url = widgetConfig.API;
    const user = this.props.userProfile;
    const interpolatedUrl = Helper.getParamsEncoded(url, user);
    // interpolatedUrl = url && user && user.role && url.replace("__orgId__", user.organization.id).replace("__emailId__", user.email).replace("__role__", user.role.name);
    Http.get(interpolatedUrl, null, null, this.props.showNotification, null, "Unable to complete request!")
      .then(response => {
        // console.log(response);
        const filters = {...this.props.currentFilters};
        filters.orgId = user.organization.id;
        filters.emailId = user.email;
        filters.role = user.role && user.role.name;
        if (response.body && response.body.errors && response.body.errors.length > 0) {
          this.props.dispatchFilter(filters);
          /* eslint-disable no-throw-literal */
          throw "error";
        }
        const data = response.body && response.body.data;
        this.setState({
          data: {
            claim: data.claimsInfo,
            brand: data.brandsInfo,
            user: data.usersInfo,
            claimsByType: data.reportedClaimsType && data.reportedClaimsType,
            claimsByBrands: data.topReportedBrands,
            claimsByUsers: data.topReporters
          },
          fetchComplete: true
        });
        filters["widget-claims-by-type"] = {dateRange: "last30days"};
        filters["widget-claims-by-brand"] = {dateRange: "last30days"};
        filters["widget-claims-by-user"] = {dateRange: "last30days"};
        this.props.dispatchFilter(filters);
      })
      .catch(() => {
        this.setState({
          error: true,
          fetchComplete: true
        });
      });
      const mixpanelPayload = { WORK_FLOW: "MY_DASHBOARD"};
      mixpanel.trackEvent(MIXPANEL_CONSTANTS.VIEW_DASHBOARD_WORKFLOW.VIEW_DASHBOARD, mixpanelPayload);
      
    }

  render() {
    return (
      <div className="c-Dashboard h-100">
        <DashboardHeader userInfo={this.props.userProfile}/>
        {!this.state.error ? <WidgetContainer
          authConfig={AUTH_CONFIG}
          currentFilters={this.props.currentFilters}
          data={this.state.data}
          fetchComplete={this.state.fetchComplete}
          widgets={widgetConfig.WIDGETS}
          userProfile={this.props.userProfile}
          widgetCommon={widgetConfig.WIDGETCOMMON}
          widgetStack={widgetConfig.WIDGETTYPES}
        /> : <GenericErrorPage generic />}
      </div>
    );
  }
}

Dashboard.propTypes = {
  currentFilters: PropTypes.object,
  dispatchFilter: PropTypes.func,
  showNotification: PropTypes.func,
  userProfile: PropTypes.object
};

const mapDispatchToProps = {
  dispatchFilter,
  showNotification
};

const mapStateToProps = state => {
  return {
    currentFilters: state.dashboard.filter,
    userProfile: state.user.profile
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
