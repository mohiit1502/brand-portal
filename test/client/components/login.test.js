import currentFilters from "../mocks/currentFilters";
import profile from "../mocks/userProfile";
import {testStore} from "../../../src/client/utility/TestingUtils";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import Dashboard from "../../../src/client/components/Dashboard";
import Http from "../../../src/client/utility/Http";
import dashboardData from "../mocks/dashboardData";
import React, {useRef} from "react";
import {Provider} from "react-redux";
import Login from "../../../src/client/components/login/login";
import LOGINCONFIG from "../../../src/client/config/contentDescriptors/landingPageTiles";

let store;

const setUp = () => {
  store = testStore({});
  return renderer.create(<Provider store={store}><BrowserRouter><Login /></BrowserRouter></Provider>);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});

describe("login page test container", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });

  describe("login Page renders without error", () => {
    it("should render the login page successfully", () => {
      jest.spyOn(Http, "get")
        .mockImplementation(() => Promise.resolve({body: {projectToken:"test-token",enableTracking:true}}))
        .mockImplementation(() => Promise.resolve({body: LOGINCONFIG}));
      const mRef = {current: document.createElement("div")};
      useRef.mockReturnValue(mRef);
      wrapper = setUp();
      const tree = wrapper.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render the login page with api failures successfully", () => {
      jest.spyOn(Http, "get")
        .mockImplementation(() => Promise.resolve({body: {}}))
        .mockImplementation(() => Promise.resolve({body: {}}));
      const mRef = {current: document.createElement("div")};
      useRef.mockReturnValue(mRef);
      wrapper = setUp();
      const tree = wrapper.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
