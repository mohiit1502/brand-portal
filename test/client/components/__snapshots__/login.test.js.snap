// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page test container login Page renders without error should render the login page successfully 1`] = `
<div
  className="login-container view-container"
>
  <nav
    className="c-LoginHeader navbar navbar-expand-lg navbar-dark login-header-nav"
  >
    <a
      className="navbar-brand walmart-brand"
      href="#"
    >
      <img
        src="wmbplightLogo.svg"
      />
    </a>
    <button
      aria-controls="collapsible-header"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#collapsible-header"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse navbar-collapsible-header"
      id="collapsible-header"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item px-4"
        >
          <a
            className="nav-link register-button"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
        <li
          className="mx-2 nav-item middle-list"
        />
        <li
          className="nav-item pl-4 d-flex align-items-center"
        >
          <a
            className="nav-link login-button"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="c-Hero"
  >
    <div
      className="row"
    >
      <div
        className="col c-Hero__left-container py-5 align-content-lg-center"
      >
        <div
          className="c-Hero__tag-line-container align-content-lg-center py-5 mx-5 my-3"
        >
          <div
            className="c-Hero__tag-line px-5 mx-5"
          >
            Helping you protect your intellectual property on Walmart.com
          </div>
          <div
            className="button-con"
          >
            <a
              className="c-Hero__register-btn btn btn-primary px-5"
              href="/register"
              onClick={[Function]}
            >
              Register
            </a>
          </div>
        </div>
      </div>
      <div
        className="col c-Hero__right-container"
      />
    </div>
  </div>
  <div
    className="c-TilesContainer row display-flex mx-0"
  />
  <div
    className="c-LoginFaq"
    id="LoginFAQ"
  >
    <div
      className="c-LoginFaq__header h4"
    >
      Frequently Asked Questions
    </div>
    <div
      className="c-LoginFaq__content"
    >
      <div
        className="c-FaqSingle"
        id="loginfaq-0"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              Why Walmart Brand Portal?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-1"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              How can I check the status of my Walmart Brand Portal application?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-2"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              What type of claims can I submit using Walmart Brand Portal?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-3"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              Is Walmart Brand Portal available only in the United States?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-4"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              Are WIPO or EUIPO trademarks accepted by Walmart Brand Portal?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-5"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              How do I submit intellectual property claims if I don't have a Walmart Brand Portal account?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-6"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              Can I use my Seller Center account credentials to login to the Brand Portal?
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c-Footer"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <nav
          className="navbar navbar-expand-lg login-footer-nav"
        >
          <div
            className="c-Footer__logoSocialContainer"
          >
            <img
              className="c-Footer__wmWhiteLogo"
              src="WMWhite-horizontal.svg"
            />
          </div>
          <div
            className="collapse navbar-collapse navbar-collapsible-footer"
            id="collapsible-footer"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <li
                className="nav-item mx-1 d-flex align-items-center"
              >
                <a
                  className="nav-link btn login-btn py-1"
                  href="/login"
                  onClick={[Function]}
                >
                  Login
                </a>
              </li>
              <li
                className="nav-item ml-1"
              >
                <a
                  className="nav-link"
                  href="/register"
                  onClick={[Function]}
                >
                  Register
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="c-Footer__copyrightContainer"
        >
          <span
            className="c-Footer__copyrightText float-left"
          >
            Â© 2021 Walmart. All rights reserved.
            <a
              className="c-Footer__privacyNav float-right"
              href="https://corporate.walmart.com/privacy-security"
              target="_blank"
            >
              Privacy Policy
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`login page test container login Page renders without error should render the login page with api failures successfully 1`] = `
<div
  className="login-container view-container"
>
  <nav
    className="c-LoginHeader navbar navbar-expand-lg navbar-dark login-header-nav"
  >
    <a
      className="navbar-brand walmart-brand"
      href="#"
    >
      <img
        src="wmbplightLogo.svg"
      />
    </a>
    <button
      aria-controls="collapsible-header"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#collapsible-header"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse navbar-collapsible-header"
      id="collapsible-header"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item px-4"
        >
          <a
            className="nav-link register-button"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
        <li
          className="mx-2 nav-item middle-list"
        />
        <li
          className="nav-item pl-4 d-flex align-items-center"
        >
          <a
            className="nav-link login-button"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="c-Hero"
  >
    <div
      className="row"
    >
      <div
        className="col c-Hero__left-container py-5 align-content-lg-center"
      >
        <div
          className="c-Hero__tag-line-container align-content-lg-center py-5 mx-5 my-3"
        >
          <div
            className="c-Hero__tag-line px-5 mx-5"
          >
            Helping you protect your intellectual property on Walmart.com
          </div>
          <div
            className="button-con"
          >
            <a
              className="c-Hero__register-btn btn btn-primary px-5"
              href="/register"
              onClick={[Function]}
            >
              Register
            </a>
          </div>
        </div>
      </div>
      <div
        className="col c-Hero__right-container"
      />
    </div>
  </div>
  <div
    className="c-TilesContainer row display-flex mx-0"
  />
  <div
    className="c-LoginFaq"
    id="LoginFAQ"
  >
    <div
      className="c-LoginFaq__header h4"
    >
      Frequently Asked Questions
    </div>
    <div
      className="c-LoginFaq__content"
    >
      <div
        className="c-FaqSingle"
        id="loginfaq-0"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              Why Walmart Brand Portal?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-1"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              How can I check the status of my Walmart Brand Portal application?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-2"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              What type of claims can I submit using Walmart Brand Portal?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-3"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              Is Walmart Brand Portal available only in the United States?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-4"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              Are WIPO or EUIPO trademarks accepted by Walmart Brand Portal?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-5"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              How do I submit intellectual property claims if I don't have a Walmart Brand Portal account?
            </button>
          </div>
        </div>
      </div>
      <div
        className="c-FaqSingle"
        id="loginfaq-6"
      >
        <div
          className="c-Accordion"
        >
          <div
            className="c-Accordion__button-container"
            data-test="accordion-buttonPanel"
          >
            <button
              className="c-Accordion__button"
              data-test="accordion-button"
              onClick={[Function]}
            >
              Can I use my Seller Center account credentials to login to the Brand Portal?
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c-Footer"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <nav
          className="navbar navbar-expand-lg login-footer-nav"
        >
          <div
            className="c-Footer__logoSocialContainer"
          >
            <img
              className="c-Footer__wmWhiteLogo"
              src="WMWhite-horizontal.svg"
            />
          </div>
          <div
            className="collapse navbar-collapse navbar-collapsible-footer"
            id="collapsible-footer"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <li
                className="nav-item mx-1 d-flex align-items-center"
              >
                <a
                  className="nav-link btn login-btn py-1"
                  href="/login"
                  onClick={[Function]}
                >
                  Login
                </a>
              </li>
              <li
                className="nav-item ml-1"
              >
                <a
                  className="nav-link"
                  href="/register"
                  onClick={[Function]}
                >
                  Register
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="c-Footer__copyrightContainer"
        >
          <span
            className="c-Footer__copyrightText float-left"
          >
            Â© 2021 Walmart. All rights reserved.
            <a
              className="c-Footer__privacyNav float-right"
              href="https://corporate.walmart.com/privacy-security"
              target="_blank"
            >
              Privacy Policy
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
